r/R/metadata.R

```diff
@@ -142,7 +142,8 @@ arrow_attributes <- function(x, only_top_level = FALSE) {
   }
 
   columns <- NULL
-  if (is.list(x) && !inherits(x, "POSIXlt")) {
+  attempt_to_save_row_level <- getOption("arrow.preserve_row_level_metadata", FALSE)
+  if (attempt_to_save_row_level && is.list(x) && !inherits(x, "POSIXlt")) {
     # However, if we are inside of a dplyr collection (including all datasets),
     # we cannot apply this row-level metadata, since the order of the rows is
     # not guaranteed to be the same, so don't even try, but warn what's going on
```

Comment:
If we disable any writing of this data at all, yes we could. Though if we retain the ability to write it with an option, we should still keep this since datasets + row-level metadata simply does not work in any circumstance.